src/lib/storage/constraints/base_constraint_checker.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <tuple>
+
+#include "storage/chunk.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+static constexpr std::tuple<bool, ChunkID> CO_TUPLE {true, MAX_CHUNK_ID};
+
+/**
+ * Base class for a constraint checker. Takes a table and a constraint definition and
+ * allows to check if the constraint is valid / will be still valid with certain values inserted.
+ */
+class BaseConstraintChecker {
+ public:
+  BaseConstraintChecker(const Table& table, const TableConstraintDefinition& constraint)
+      : _table(table), _constraint(constraint) {}
+  virtual ~BaseConstraintChecker() = default;
+
+  /**
+   * Checks if a constraint is satisfied on a table. For consitency with the other function, this function returns also
+   * an CommitID. Namely, it returns MAX_COMMIT_ID, as the second value, to indicate that no chunk can be skipped if a
```

Comment:
But it does *not* return a commit id. You returning MAX_COMMIT_ID without anything breaking is a sign that we might be better off without that consistency.

Reply:
Agreed. I think `MAX_COMMIT_ID` was meant to say `MAX_CHUNK_ID`, but like you said `is_valid` does not have to return a tuple.